Sources/Views/Cells/AudioMessageCell.swift

```diff
@@ -68,27 +82,37 @@ open class AudioMessageCell: MessageContentCell {
         playButton.addConstraints(left: messageContainerView.leftAnchor, centerY: messageContainerView.centerYAnchor, leftConstant: 5)
         activityIndicatorView.addConstraints(centerY: playButton.centerYAnchor, centerX: playButton.centerXAnchor)
         durationLabel.addConstraints(right: messageContainerView.rightAnchor, centerY: messageContainerView.centerYAnchor, rightConstant: 15)
-        progressView.addConstraints(left: playButton.rightAnchor, right: durationLabel.leftAnchor, centerY: messageContainerView.centerYAnchor, leftConstant: 5, rightConstant: 5)
+        slider.addConstraints(left: playButton.rightAnchor, right: durationLabel.leftAnchor, centerY: messageContainerView.centerYAnchor, leftConstant: 10, rightConstant: 5)
     }
 
     open override func setupSubviews() {
         super.setupSubviews()
         messageContainerView.addSubview(playButton)
         messageContainerView.addSubview(activityIndicatorView)
         messageContainerView.addSubview(durationLabel)
-        messageContainerView.addSubview(progressView)
+        messageContainerView.addSubview(slider)
         setupConstraints()
     }
 
     open override func prepareForReuse() {
         super.prepareForReuse()
-        progressView.progress = 0
+        slider.value = 0
         playButton.isSelected = false
         activityIndicatorView.stopAnimating()
         playButton.isHidden = false
         durationLabel.text = "0:00"
     }
 
+    open override func gestureRecognizerShouldBegin(_ gestureRecognizer: UIGestureRecognizer) -> Bool {
+        let touchPoint = gestureRecognizer.location(in: messageContainerView)
+
+        if slider.frame.contains(touchPoint) {
+            return false
+        }
+
+        return super.gestureRecognizerShouldBegin(gestureRecognizer)
```

Comment:
Because in `MessagesCollectionView.swift` `func setupGestureRecognizers` there is a global tap gesture recognizer that we must punch through so `UISlider` gets clean touches.

Reply:
I found that I wasn't getting clean sliding with `UISlider` if these were not overridden. 

Also otherwise just simply tapping the `UISlider` could count as a message tap/long press.

Finally it is good form to just let UISlider do its own thing and not be affected by any other gesture recognizers the user may place throughout the view hierarchy.